src/spikeinterface/core/sparsity.py

```diff
@@ -605,43 +609,55 @@ def estimate_sparsity(
         chan_locs = recording.get_channel_locations()
         probe = recording.create_dummy_probe_from_locations(chan_locs)
 
-    nbefore = int(ms_before * recording.sampling_frequency / 1000.0)
-    nafter = int(ms_after * recording.sampling_frequency / 1000.0)
-
-    num_samples = [recording.get_num_samples(seg_index) for seg_index in range(recording.get_num_segments())]
-    random_spikes_indices = random_spikes_selection(
-        sorting,
-        num_samples,
-        method="uniform",
-        max_spikes_per_unit=num_spikes_for_sparsity,
-        margin_size=max(nbefore, nafter),
-        seed=2205,
-    )
-    spikes = sorting.to_spike_vector()
-    spikes = spikes[random_spikes_indices]
-
-    templates_array = estimate_templates_with_accumulator(
-        recording,
-        spikes,
-        sorting.unit_ids,
-        nbefore,
-        nafter,
-        return_scaled=False,
-        job_name="estimate_sparsity",
-        **job_kwargs,
-    )
-    templates = Templates(
-        templates_array=templates_array,
-        sampling_frequency=recording.sampling_frequency,
-        nbefore=nbefore,
-        sparsity_mask=None,
-        channel_ids=recording.channel_ids,
-        unit_ids=sorting.unit_ids,
-        probe=probe,
-    )
+    if method != "by_property":
```

Comment:
Would it be possible to add a little more to the docstring here, I don't quite understand the use for `by_property`. I am used to a property per channel e.g. "group" returns two recordings split by channel group. I can't align that in my head with e.g.  `sorting_analyzer.sorting.get_property(by_property)[unit_ind]` the property is index by unit not channel. So I am confused 😅

Reply:
so we have a `run_sorter_by_property` function. In that case, a unit is assigned a property value corresponding to the property group it was found on (e.g. `tetrode0`)

Reply:
I think it might make sense to add the run sorter by group example for sure!